File: 2.Arrays/readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<math.h>

int colemanLiauFormula(int letters, int words, int sentences); //function decleration
void outputFromFormula(int grade); //function decleration

int main(void)
{
	//counters
	int lettersCounter = 0;
	int wordsCounter = 0;
	int sentencesCounter = 0;
	//getting input from user
	string text = get_string("Enter your text : ");
	int textLength = strlen(text);
	//algorithm for counting
	for (int i = 0, lettersBetweenSpaces = 0, wordsBetweenSentences = 0; i < textLength; i++)
	{
		if (((65 <= (int)text[i]) && ((int)text[i] <= 90)) || ((97 <= (int)text[i])
			&& ((int)text[i] <= 122)))  // letters counter ++, lettersbetweenspaces ++
		{
			lettersCounter++;
			lettersBetweenSpaces++;
		}
		if ((int)text[i] == 32)  // words counter ++ , words between sentences ++ ,lettersBetweenSpaces initialize
		{
			if (lettersBetweenSpaces >= 1)
			{
				wordsCounter++;
				wordsBetweenSentences++;

			}
			lettersBetweenSpaces = 0;
		}
		// ((int)text[i] == 59)
		if (((int)text[i] == 46) || ((int)text[i] == 33) || ((int)text[i] == 63))  // sentences counter ++, wordsbetweensentences initialize
		{
			if (lettersBetweenSpaces >= 1)
			{
				sentencesCounter++;
				wordsCounter++;
			}
			wordsBetweenSentences = 0;
			lettersBetweenSpaces = 0;

		}

	}
	// printf("Letters --> %i , Words --> %i , Sentences --> %i\n", lettersCounter, wordsCounter, sentencesCounter);
	outputFromFormula(colemanLiauFormula(lettersCounter, wordsCounter,
		sentencesCounter)); //call output function and pass the formula returtning value to it.

}

int colemanLiauFormula(int letters, int words, int sentences)  //little math
{
	int numberOfLettersPer100W = 100 / (float)words * letters;
	int numberOfSentencesPer100 = 100 / (float)words * sentences;
	int index = round((0.0588 * numberOfLettersPer100W) - (0.296 * numberOfSentencesPer100) - 15.8);
	return index;
}

void outputFromFormula(int grade) // if tree
{
	if (grade < 1)
	{
		printf("Before Grade 1\n");
	}
	else if (grade > 16)
	{
		printf("Grade 16+\n");
	}
	else
	{
		printf("Grade %i\n", grade);
	}
}

File: 6.Python/readability.py
def luhn_formula(letters, words, sentences):
    return (0.0588 * (100/words*letters)) - (0.296 * (100/words*sentences)) - 15.8


def output(index):
    if index < 1:
        print("Before Grade 1\n")
    if index > 16:
        print("Grade 16+\n")
    else:
        print(f"Grade {index}\n")


def main():
    text = input("Enter your text -- > ")
    # check for content vetween space / / ? !
    letters_between_words = 0
    # counters
    letters_counter = 0
    words_counter = 0
    sentences_counter = 0

    for char in text:
        # a-z A-Z
        if(ord(char.upper()) >= 65 and ord(char.upper()) <= 90):
            letters_counter += 1
            letters_between_words += 1
        # space
        if(ord(char) == 32):
            if(letters_between_words >= 1):
                words_counter += 1
                letters_between_words = 0
        # ? ! .
        if(ord(char) in [46, 33, 63]):
            if (letters_between_words >= 1):
                sentences_counter += 1
                words_counter += 1
                letters_between_words = 0
    output(round(luhn_formula(letters_counter, words_counter, sentences_counter)))


main()